consensus/types/src/fork_versioned_response.rs

```diff
@@ -0,0 +1,95 @@
+use crate::ForkName;
+use serde::de::DeserializeOwned;
+use serde::{Deserialize, Deserializer, Serialize};
+use serde_json::value::Value;
+
+#[derive(Debug, PartialEq, Clone, Serialize, Deserialize)]
+pub struct ExecutionOptimisticForkVersionedResponse<T> {
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub version: Option<ForkName>,
+    pub execution_optimistic: Option<bool>,
+    pub data: T,
```

Comment:
don't we also want `T: ForkVersionDeserialize` ?

Reply:
done! although we actually wanted this to be less restrictive but now `Deserialize` is only implemented for `T: ForkVersionedDeserialize` types for either of these :)